### DISCUSSION TITLE
How does GitHub Pages detect pull requests to be auto-closed

### DISCUSSION THREAD

**Discussion by oharakouhei** [community/community#22579] [Other Feature Feedback, Questions, & Ideas]
<p>Hello.</p>
<p>How does Github Pages detect pull requests to be auto-closed?</p>
<p>I think that it is the same processing as <code>[git branch --merged](https://github.com/git/git/blob/master/builtin/branch.c#L110)</code>, is it right?</p>
<p>Some pull requests is auto-closed.</p>
<p>Some cases like Example2 does not auto-close pull request.</p>
<h2>Example1: <code>&lt;font color="#000000"&gt;add_aaa</code> is auto-closed </h2>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/c503ca9f39af6a8822b75cace9e1b42aa3bb9b31.png" data-download-href="/uploads/short-url/s6SbsNTyVcdjEIQluRJJiqbcp7b.png?dl=1" title="スクリーンショット 2018-12-01 16.48.31.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/8342518/181072061-b8f3632b-17bb-419f-8acc-e7c80f97f42d.png" alt="スクリーンショット 2018-12-01 16.48.31.png" data-base62-sha1="s6SbsNTyVcdjEIQluRJJiqbcp7b" width="690" height="161" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/c503ca9f39af6a8822b75cace9e1b42aa3bb9b31_2_690x161.png, https://aws1.discourse-cdn.com/github/original/1X/c503ca9f39af6a8822b75cace9e1b42aa3bb9b31.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/c503ca9f39af6a8822b75cace9e1b42aa3bb9b31.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/c503ca9f39af6a8822b75cace9e1b42aa3bb9b31_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">スクリーンショット 2018-12-01 16.48.31.png</span><span class="informations">870×204 15.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I create two pull requests</p>
<ul>
<li>develop (base: master)</li>
<li>add_aaa (base: master)</li>
</ul>
<p>If I merge develop to master, add_aaa is auto-closed </p>
<p>  </p>
<h2>Example2: <code>&lt;font color="#000000"&gt;add_aaa</code> is <strong>not</strong> auto-closed </h2>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/4d7ca2203c23fff310ef8988cca9fa791c84a21b.png" data-download-href="/uploads/short-url/b3tN218VuYT0JfccmJ4vhx9Y5nB.png?dl=1" title="スクリーンショット 2018-12-01 17.01.34.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/8342518/181072062-ccc8317c-9139-42ce-a5bc-19854a71aa71.png" alt="スクリーンショット 2018-12-01 17.01.34.png" data-base62-sha1="b3tN218VuYT0JfccmJ4vhx9Y5nB" width="690" height="218" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/4d7ca2203c23fff310ef8988cca9fa791c84a21b_2_690x218.png, https://aws1.discourse-cdn.com/github/optimized/1X/4d7ca2203c23fff310ef8988cca9fa791c84a21b_2_1035x327.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/4d7ca2203c23fff310ef8988cca9fa791c84a21b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/4d7ca2203c23fff310ef8988cca9fa791c84a21b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">スクリーンショット 2018-12-01 17.01.34.png</span><span class="informations">1176×372 28.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I create two pull requests same as Example1.</p>
<p>Even if I merge develop to master, add_aaa is <strong>not</strong> auto-closed.</p>
<p>I think the reason is that you can not trace the commit graph from the master (after develop is merged) to the head of the add_aaa branch.</p>
<p>Thanks in advance for your help!</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/oharakouhei">@oharakouhei</a>,</p>
<p>Thanks for being here! You are correct! here is more information on <a href="https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/" rel="nofollow noopener">configuring a publishing source for GitHub pages.</a> Thanks for the cool graphs btw!</p>
<p>Cheers,</p>
<p>Andrea</p>
